index.d.ts

```diff
@@ -8,17 +8,15 @@
 // Usage:
 // const mockStore = configureStore();
 
-declare module "redux-mock-store" {
-    import {Store} from 'redux';
+import {Store, Middleware, Action, Unsubscribe} from 'redux';
 
-    interface MockStore extends Store {
-        getState(): any;
-        getActions(): Array<any>;
-        dispatch(action: any): any;
-        clearActions(): void;
-        subscribe(): any;
-    }
-
-    function configureStore(...args: any[]) : (...args: any[]) => MockStore;
-    export = configureStore;
+interface MockStore<S> extends Store<S> {
+    getState():S;
+    getActions():Array<Action>;
+    dispatch(action:Action):any;
+    clearActions():void;
+    subscribe(listener: Function):Unsubscribe;
 }
+
+declare function configureStore(...middlewares:Middleware[]):(...args:any[]) => MockStore;
```

Comment:
Here throws same error => _Generic type 'MockStore\<S\>' requires 1 type argument(s)_.

One solution could be:

`declare function configureStore(...middlewares:Middleware[]):(...args:any[]) => MockStore<any>;`

What do you think?